EIPS/eip-7701.md

```diff
@@ -109,6 +116,18 @@ If it is the first code section of a contract, it can act as an entry point duri
 Only a single section per role is allowed in a contract.
 This rule is validated during contract creation.
 
+### Context variable for the `entry_point_role` value
+
+During the execution of the `Sender`, `Paymaster` or a `Deployer` code as defined by the `AA_TX_TYPE` transaction,
+the global `entry_point_role` variable is set to the corresponding role.
```

Comment:
How is this variable accessed from EVM?  A new opcode?

Reply:
This variable is not actually visible inside the EVM. It is used by the EVM implementation to decide which code section to enter when using `DELEGATECALL`.

Reply:
We could encode that variable into `ORIGIN` so instead of `0x7701000000000000000000000000000000007701` it would return `0x7701000000000000000000000000000000000000 | entry_point_role` but I'm not sure if we should.  May be useful if a function called by the role's code section needs to know the context. OTOH the code itself could pass it as an arg if it's needed.